librz/core/cmd/cmd_analysis.c

```diff
@@ -7061,6 +6811,127 @@ RZ_IPI RzCmdStatus rz_analysis_function_cc_reg_usage_handler(RzCore *core, int a
 	return RZ_CMD_STATUS_OK;
 }
 
+RZ_IPI RzCmdStatus rz_analysis_function_describe_offset_handler(RzCore *core, int argc, const char **argv) {
+	RzList *list = rz_analysis_get_functions_in(core->analysis, core->offset);
+	if (rz_list_empty(list)) {
+		RZ_LOG_ERROR("No function found in 0x%08" PFMT64x ".\n", core->offset);
+		rz_list_free(list);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	RzListIter *it;
+	RzAnalysisFunction *fcn;
+	rz_list_foreach (list, it, fcn) {
+		st64 delta = core->offset - fcn->addr;
+		if (delta > 0) {
+			rz_cons_printf("%s + %" PFMT64d "\n", fcn->name, delta);
+		} else if (delta < 0) {
+			rz_cons_printf("%s - %" PFMT64d "\n", fcn->name, -delta);
+		} else {
+			rz_cons_printf("%s\n", fcn->name);
+		}
+	}
+	rz_list_free(list);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_function_add_recu_handler(RzCore *core, int argc, const char **argv) {
+	const char *name = argc == 2 ? argv[1] : NULL;
+	bool analyze_recursively = true;
+	if (!strcmp(argv[0], "af")) {
+		analyze_recursively = rz_config_get_i(core->config, "analysis.calls");
+	}
+	bool result = rz_core_analysis_function_add(core, name, core->offset, analyze_recursively);
+	return result ? RZ_CMD_STATUS_OK : RZ_CMD_STATUS_ERROR;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_function_create_handler(RzCore *core, int argc, const char **argv) {
+	RzAnalysisDiff *diff = rz_analysis_diff_new();
+	if (!diff) {
+		RZ_LOG_ERROR("Cannot init RzAnalysisDiff\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	if (argc == 4) {
+		switch (argv[3][0]) {
+		case 'm':
+			diff->type = RZ_ANALYSIS_DIFF_TYPE_MATCH;
+			break;
+		case 'u':
+			diff->type = RZ_ANALYSIS_DIFF_TYPE_UNMATCH;
+			break;
+		default:
+			break;
+		}
+	}
+	int type = RZ_ANALYSIS_FCN_TYPE_FCN;
+	if (argc >= 3) {
+		switch (argv[2][0]) {
+		case 'l':
+			type = RZ_ANALYSIS_FCN_TYPE_LOC;
+			break;
+		case 'i':
+			type = RZ_ANALYSIS_FCN_TYPE_IMP;
+			break;
+		case 's':
+			type = RZ_ANALYSIS_FCN_TYPE_SYM;
+			break;
+		default:
+			break;
+		}
+	}
+	RzAnalysisFunction *fcn = rz_analysis_create_function(core->analysis, argv[1], core->offset, type, diff);
+	if (!fcn) {
+		RZ_LOG_ERROR("Cannot add function (duplicated)\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_function_del_handler(RzCore *core, int argc, const char **argv) {
+	RzAnalysisFunction *f = analysis_get_function_in(core->analysis, core->offset);
+	if (!f) {
+		return RZ_CMD_STATUS_ERROR;
```

Comment:
Probably worth printing the error message as well.

Reply:
`analysis_get_function_in` does it